sysfs/class_infiniband.go

```diff
@@ -251,7 +251,7 @@ func parseInfiniBandCounters(portPath string) (*InfiniBandCounters, error) {
 		name := filepath.Join(path, f.Name())
 		value, err := util.SysReadFile(name)
 		if err != nil {
-			if os.IsNotExist(err) || err.Error() == "operation not supported" || err.Error() == "invalid argument" {
+			if os.IsNotExist(err) || err.Error() == "operation not supported" || err.Error() == "invalid argument" || err.Error() == "permission denied" {
```

Comment:
```suggestion
			if os.IsNotExist(err) || os.IsPermission(err) || err.Error() == "operation not supported" || err.Error() == "invalid argument" {
```

Reply:
Yes, let's use the `os.IsPermission(err)` function.

Reply:
Updated to use `os.IsPermission(err)`